### DISCUSSION TITLE
Any tutorial to get started ?

### DISCUSSION THREAD

**Discussion by yxdunc** [rust-av/rust-av#176] [General]
I'm looking into using rust-av for my first video processing project and was hoping I could find an introduction on how to use the lib.
If there isn't any tutorial yet I would be happy to contribute but I would still need someone to point me in the right directions :)

**Comment by lu-zero** [comment]
Everything is in current flux, hopefully we'll have some tutorial as soon as the new API is finalised :)

**Comment by yxdunc** [comment]
Thanks for your answer! What's the approximative timeline ? Anyway to track the progress ?
Is it possible to help ? :)

**Reply by lu-zero** [reply]
Help is surely welcome, right now @Luni-4 is experimenting a lot and might enjoy some feedback. I'll be back to it myself in few weeks (right now I have some work related endeavors that require my full attention)

**Reply by yxdunc** [reply]
Alright thanks, let me know Luni-4 if I can be of any help!

**Reply by Luni-4** [reply]
Hi! Yeah, I'm experimenting quite a lot with `format` and `codec` APIs. As they are now, it's not possible to create a muxer which can either use only the `Write` trait or the `Write + Seek` traits inside the methods requested by the `Muxer` trait. Have a look at the [wav](https://github.com/rust-av/wav) crate and you will see that is quite difficult updating that to the current `format` version. 

Another problem is that we can't create a list of muxers/demuxers and coders/decoders right now because we don't `Box` the objects implementing the respective traits, so we need to expand and simplify `CodecList` and add more tests to verify its correctness. 

There are some branches right now that explores some ways to fix those problems but they are currently incomplete
